### ISSUE TITLE
TypeScript: should `emit` and `off` be a part of `Broker` namespace?

### ISSUE THREAD

**Issue by GoFightNguyen** [Foo-Foo-MQ/foo-foo-mq#44]
`foo-foo-mq` version: 8.0.0
`typescript` version: 4.4.4

We have unit tests asserting that our code correctly handles events raised by `foo-foo-mq`. here is an example
```typescript
import * as sinon from 'sinon';
import * as rabbot from 'foo-foo-mq';
import { assert } from 'chai';
import { EventEmitter } from 'events';

describe('#registerEvents specific events checks', () => {
    let sandbox: sinon.SinonSandbox;
    let eventEmitter: EventEmitter;

    beforeEach(() => {
      sandbox = sinon.createSandbox();
      eventEmitter = new EventEmitter();
      
       // our code that responsds to foo-foo-mq events by re-emitting
      registerEvents(eventEmitter, 1, 5);
    });

    afterEach(() => {
      rabbot.off();
    });

    after(() => {
      sandbox.restore();
    });

    it('expected data returned from the failed event', (done) => {
      const failureData = {
        consecutiveFailures: 1,
      };
      const timeStampObj = new Date();

      eventEmitter.on('failed', (data) => {
        assert.deepEqual(data, {
          connectionAttempts: failureData.consecutiveFailures,
          attemptsRemaining: 4,
          remainingTimeToRetry: 1, // how to calculated value
          message:
            'Attempting to reconnect until time runs out or retries are expired',
        });
        done();
      });
      rabbot.emit('failed', failureData, timeStampObj);
    });
});
```

The problem is that `tsc` fails with `error TS2339: Property 'emit' does not exist on type 'typeof Broker'.`

Should `emit` be added to the `Broker` [type/namespace](https://github.com/Foo-Foo-MQ/foo-foo-mq/blob/a8781ba72d8c40f4cbc4a3a09bdff2fc61771327/src/index.d.ts#LL5C21-L5C21)? Or is there a better way to simulate a `foo-foo-mq` event?

**Comment by GoFightNguyen** [comment]
In addition, we are experiencing the same `tsc` error for `off()`, which is used in the `afterEach()`

**Comment by zlintz** [comment]
Yes you are correct. You can see those in the `monologue.js` file. The current `index.d.ts` was graciously provided by another user but it likely was they were not using that functionality. Feel free to make a PR for this and we can get it merged in.  There is another change that also needs to get release soon so now is a good time for this to get in @GoFightNguyen

**Comment by GoFightNguyen** [comment]
For history, `monologue.js` is in reference to [`node-monologue`](https://www.npmjs.com/package/node-monologue)

**Comment by GoFightNguyen** [comment]
@zlintz Any idea when/if a release will be made soon?

**Comment by zlintz** [comment]
Thanks for checking, I just released `foo-foo-mq@8.0.1` for you
